pkg/operator2/operator.go

```diff
@@ -330,12 +344,23 @@ func (c *authOperator) handleSync(operatorConfig *operatorv1.Authentication) err
 		operatorDeployment,
 		resourceVersions...,
 	)
+
+	// redeploy on operatorConfig.spec changes or when bootstrap user is deleted
+	forceRollOut := operatorConfig.Generation != operatorConfig.Status.ObservedGeneration
+	if c.bootstrapUserChangeRollOut {
+		if userExists, err := c.bootstrapUserExists(); err != nil {
+			klog.Warningf("Unable to determine the state of bootstrap user: %v", err)
+		} else if !userExists {
+			forceRollOut = true
+			c.bootstrapUserChangeRollOut = !c.bootstrapUserChangeRollOut
```

Comment:
This can never change from `false` to `true` (given the wrapping condition), so you can safely do `c.bootstrapUserChangeRollOut = false`

Reply:
Done!